mongo_connector/doc_managers/neo4j_doc_manager.py

```diff
@@ -33,13 +31,20 @@ class DocManager(DocManagerBase):
   def __init__(self, url, auto_commit_interval=DEFAULT_COMMIT_INTERVAL,
                  unique_key='_id', chunk_size=DEFAULT_MAX_BULK, **kwargs):
     self.remote_graph = Graph(url, **kwargs.get('clientOptions', {}))
-    graph = Graph()
+    self.graph = Graph()
     self.auto_commit_interval = auto_commit_interval
     self.unique_key = unique_key
     self.chunk_size = chunk_size
     if self.auto_commit_interval not in [None, 0]:
       self.run_auto_commit()
-  
+    self._formatter = DefaultDocumentFormatter()
+    self.query_nodes = {}
+    self.doc_types = []
+
+  def apply_id_constraint(self, doc_type):
+    constraint = "CREATE CONSTRAINT ON (d:{doc_type}) ASSERT d.id IS UNIQUE".format(doc_type=doc_type)
```

Comment:
Ok, this case will also happen. But take a look at the above example. By inserting this kind of document, we would have a bunch of nested data. Not always we will have a proper id for then. Keeping it into the same node wouldn't be bad?